packages/boot/test/bootstrapTests/benchmarkerator.js

```diff
@@ -0,0 +1,579 @@
+  const h1 = `${'Stat'.padEnd(w1)}`;
+  const d1 = `${''.padEnd(w1, '-')}`;
+
+  const w2 = 6;
+  const h2 = `${'Delta'.padStart(w2)}`;
+  const d2 = ` ${''.padStart(w2 - 1, '-')}`;
+
+  const w3 = 10;
+  const h3 = `${'PerRound'.padStart(w3)}`;
+  const d3 = ` ${''.padStart(w3 - 1, '-')}`;
+
+  const cpr = pn(stats.cranksPerRound).trim();
+  log(
+    `${stats.cranks} cranks over ${stats.rounds} rounds (${cpr} cranks/round)`,
+  );
+  if (stats.cranks > 0) {
+    log(
+      // prettier-ignore
+      `${stats.cranks} cranks in ${stats.elapsedTime}ns (${stats.timePerCrank.toFixed(3)}/crank})`,
+    );
+  }
+  log(
+    // prettier-ignore
+    `${stats.rounds} rounds in ${stats.elapsedTime}ns (${stats.timePerRound.toFixed(3)}/round})`,
+  );
+  log(`${h1} ${h2} ${h3}`);
+  log(`${d1} ${d2} ${d3}`);
+
+  const data = stats.data;
+  for (const [key, entry] of Object.entries(data)) {
+    const col1 = `${key.padEnd(w1)}`;
+    const col2 = `${String(entry.delta).padStart(w2)}`;
+    const col3 = `${pn(entry.deltaPerRound).padStart(w3)}`;
+    log(`${col1} ${col2} ${col3}`);
+  }
+};
+
+const organizeRoundsStats = (
+  rawBefore,
+  rawAfter,
+  elapsedTime,
+  cranks,
+  rounds,
+) => {
+  const stats = {
+    elapsedTime,
+    cranks,
+    rounds,
+    timePerCrank: cranks ? elapsedTime / cranks : 0,
+    timePerRound: elapsedTime / rounds,
+    cranksPerRound: cranks / rounds,
+    data: {},
+  };
+
+  // Note: the following assumes rawBefore and rawAfter have the same keys.
+  for (const [key, value] of Object.entries(rawBefore)) {
+    if (isMainKey(key)) {
+      const delta = rawAfter[key] - value;
+      stats.data[key] = {
+        delta,
+        deltaPerRound: delta / rounds,
+      };
+    }
+  }
+  return stats;
+};
+
+/**
+ * Produce a new benchmarkerator object.
+ *
+ * Note: Most likely you do not need to use this, as this the benchmarkerator is
+ * typically a singleton.  But we provide it just in case...
+ *
+ * @returns {Promise<Benchmarkerator>}
+ */
+export const makeBenchmarkerator = async () => {
+  const benchmarks = new Map();
+
+  const setupStartTime = readClock();
+  const swingsetTestKit = await makeStandaloneSwingsetTestKit(undefined, {
+    defaultManagerType,
+    verbose,
+  });
+  const {
+    runUtils,
+    storage: chainStorage,
+    shutdown,
+    controller,
+    getCrankNumber,
+  } = swingsetTestKit;
+  const { EV } = runUtils;
+
+  // Wait for ATOM to make it into agoricNames
+  // E(bootstrap).consumeItem('vaultFactoryKit');
+  await EV.vat('bootstrap').consumeItem('vaultFactoryKit');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes =
+    makeAgoricNamesRemotesFromFakeStorage(chainStorage);
+
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    chainStorage,
+    agoricNamesRemotes,
+  );
+
+  const actors = {
+    gov1: await walletFactoryDriver.provideSmartWallet(
+      'agoric1ldmtatp24qlllgxmrsjzcpe20fvlkp448zcuce',
```

Comment:
are these constants meaningful? please source from a constants block or module

Reply:
Meh.  They're pretty arbitrary.  I think this is one of those things that's still in discovery mode.